src/main/java/org/radarcns/management/service/SubjectService.java

```diff
@@ -324,19 +301,52 @@ public MinimalSourceDetailsDTO assignOrUpdateSource(Subject subject, SourceType
             log.error("Cannot find assigned source with sourceId or a source of sourceType with "
                     + "the specified producer and model is already registered for subject login ");
             Map<String, String> errorParams = new HashMap<>();
-            errorParams.put("message", "Cannot find assigned source with sourceId or a source of "
-                    + "sourceType with the specified producer and model is already registered "
-                    + "for subject login ");
             errorParams.put("producer", sourceType.getProducer());
             errorParams.put("model", sourceType.getModel());
             errorParams.put("subject-id", subject.getUser().getLogin());
             errorParams.put("sourceId", sourceRegistrationDto.getSourceId().toString());
-            throw new CustomParameterizedException("InvalidRequest", errorParams);
+            throw new BadRequestException("Cannot find assigned source with sourceId or a source "
```

Comment:
Is this not a conflictexception? E.g. http status code 409?

Reply:
This cannot be a conflict request, since we did not get enough data to process this request. This is a situation where we cannot both create a new source, or update source.

Reply:
Ok understood